### PR TITLE
Googet clean updates

### PR DESCRIPTION
**Pull request by adjackura**
Add the -all flag to googet clean to clean the entire cache directory
Add the -packages flag to googet clean to clean the cache for a specific package
Add tests for googet_clean functions

### CODE REVIEW

File: googet_clean.go [hunk 1]
```diff
@@ -18,44 +18,84 @@ import (
...
+func (cmd *cleanCmd) SetFlags(f *flag.FlagSet) {
+	f.BoolVar(&cmd.all, "all", false, "clear out the entire cache directory")
+	f.StringVar(&cmd.packages, "packages", "", "comma separated list of packages to clear out of the cache")
+}
 
 func (cmd *cleanCmd) Execute(_ context.Context, _ *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
-	fmt.Println("Removing all files and directories in cachedir that dont correspond to a currently installed package.")
+	if cmd.all {
+		fmt.Println("Removing all files and directories in cachedir.")
+		clean([]string{})
```
Comment (jrunningen) [inline googet_clean.go]: I think you can pass nil instead of []string{}.

File: googet_clean.go [hunk 2]
```diff
@@ -18,44 +18,84 @@ import (
...
+	} else if cmd.packages != "" {
+		pl := strings.Split(cmd.packages, ",")
+		fmt.Printf("Removing package cache for %s\n", pl)
+		cleanPackages(pl)
+	} else {
+		fmt.Println("Removing all files and directories in cachedir that dont correspond to a currently installed package.")
```
Comment (jrunningen) [inline googet_clean.go]: "don't"

Comment (jrunningen) [inline googet_clean.go]: "don't"

File: googet_clean.go (hunk 1 above)
Comment (adjackura) [inline googet_clean.go]: done

**Comment by jrunningen** [comment]
LGTM

**Pull request merged**